src/Chapter3.hs

```diff
@@ -907,9 +1002,30 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+newtype Gold = Gold Int deriving (Show)
+
+data List a
+  = Empty
+  | Cons a (List a)
+  deriving (Show)
+
 class Append a where
-    append :: a -> a -> a
+  append :: a -> a -> a
 
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append (Gold n1) (Gold n2) = Gold (n1 + n2)
+
+instance Append (List a) where
+  append :: List a -> List a -> List a
+  append Empty ys       = ys
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
append xs Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤